### PR TITLE
Remove some unnecessary dependency injection

### PR DESCRIPTION
**Pull request by seanmonstar**
Just doing some spring cleaning (I do have an end goal in mind, but must get there piece by piece).

### CODE REVIEW

File: test/test_mailer_server.js [hunk 1]
```diff
@@ -2,54 +2,23 @@
...
-var P = require('../lib/promise')
-var request = require('request')
 var createDBServer = require('fxa-auth-db-mysql')
 
 function TestServer(config) {
   this.config = config
   this.server = null
 }
 
-function waitLoop(testServer, url, cb) {
```
Comment (rfk) [inline test/test_mailer_server.js]: Is this an unrelated change?  It doesn't appear related to dependency injection.  I'm fine with it, just checking!

Comment (seanmonstar) [inline test/test_mailer_server.js]: It is unrelated, so it is in another commit. It was something I noticed I had missed from the earlier remote test refactoring. Just felt :shrug: about filing a separate PR.
